File: RedBlockIOS/RedBlockIOS/Utils/ShareView.m
//
//  ShareView.m
//  RedBlockIOS
//
//  Created by che on 15/6/30.
//  Copyright (c) 2015年 cyzcj. All rights reserved.
//

#import "ShareView.h"
#import "Utils.h"
#import "ShareUtils.h"

@interface ShareView ()
{
    BOOL    isShowSettingMenu;
    
    UIView  *shareMenuView;
}
@end

@implementation ShareView

- (id)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        [self initInterface];
    }
    return self;
}

- (void)initInterface {
    self.backgroundColor = [UIColor clearColor];
    self.clipsToBounds = YES;
    self.layer.cornerRadius = 24;
    
    shareMenuView = [[UIView alloc] initWithFrame:CGRectMake(0, self.bounds.size.height, self.bounds.size.width, self.bounds.size.height)];
    shareMenuView.backgroundColor = [UIColor colorWithRed:116.f/255 green:183.f/255 blue:215.f/255 alpha:1.0f];
    shareMenuView.layer.cornerRadius = 24;
    shareMenuView.clipsToBounds = YES;
    [self addSubview:shareMenuView];
    
    UIButton    *shareButton = [[UIButton alloc] initWithFrame:CGRectMake(0, self.bounds.size.height-48, 48, 48)];
    [shareButton addTarget:self action:@selector(settingAction:) forControlEvents:UIControlEventTouchUpInside];
    [shareButton setImage:[UIImage imageNamed:@"settingBtn.png"] forState:UIControlStateNormal];
    [self addSubview:shareButton];
    
//    UIButton *faceBookBtn = [[UIButton alloc] initWithFrame:CGRectMake(4, 15, 40, 40)];
//    [faceBookBtn addTarget:self action:@selector(fbAction:) forControlEvents:UIControlEventTouchUpInside];
//    [faceBookBtn setImage:[UIImage imageNamed:@"fb.png"] forState:UIControlStateNormal];
//    [shareMenuView addSubview:faceBookBtn];
//    
//    UIButton    *tweentBtn = [[UIButton alloc] initWithFrame:CGRectMake(4, 65, 40, 40)];
//    [tweentBtn addTarget:self action:@selector(tweetAction:) forControlEvents:UIControlEventTouchUpInside];
//    [tweentBtn setImage:[UIImage imageNamed:@"tw.png"] forState:UIControlStateNormal];
//    [shareMenuView addSubview:tweentBtn];
    
    UIButton    *chatBtn = [[UIButton alloc] initWithFrame:CGRectMake(4, 15, 40, 40)];
    [chatBtn addTarget:self action:@selector(chatAction:) forControlEvents:UIControlEventTouchUpInside];
    [chatBtn setImage:[UIImage imageNamed:@"wechat-icon.png"] forState:UIControlStateNormal];
    [shareMenuView addSubview:chatBtn];
    
    UIButton    *friendBtn = [[UIButton alloc] initWithFrame:CGRectMake(4, 65, 40, 40)];
    [friendBtn addTarget:self action:@selector(friendAction:) forControlEvents:UIControlEventTouchUpInside];
    [friendBtn setImage:[UIImage imageNamed:@"friends-icon.png"] forState:UIControlStateNormal];
    [shareMenuView addSubview:friendBtn];
}

- (void)settingAction:(id)sender {
    UIButton *button = (UIButton*)sender;
    button.userInteractionEnabled = NO;
    [UIView animateWithDuration:0.5 delay:0.0 usingSpringWithDamping:0.8 initialSpringVelocity:2.0f options:UIViewAnimationOptionCurveLinear animations:^{
        shareMenuView.frame = isShowSettingMenu?CGRectMake(0, self.bounds.size.height, self.bounds.size.width, self.bounds.size.height):self.bounds;
    } completion:^(BOOL finished) {
        isShowSettingMenu = !isShowSettingMenu;
        button.userInteractionEnabled = YES;
    }];
}

- (void)fbAction:(id)sender {
    UIButton *button = (UIButton*)sender;
    button.userInteractionEnabled = NO;
    
    button.userInteractionEnabled = YES;
}

- (void)tweetAction:(id)sender {
    UIButton *button = (UIButton*)sender;
    button.userInteractionEnabled = NO;
    
    button.userInteractionEnabled = YES;
}

- (void)chatAction:(id)sender {
    UIButton *button = (UIButton*)sender;
    button.userInteractionEnabled = NO;
    [[ShareUtils shareInstance] shareToWeChat];
    button.userInteractionEnabled = YES;
}

- (void)friendAction:(id)sender {
    UIButton *button = (UIButton*)sender;
    button.userInteractionEnabled = NO;
    [[ShareUtils shareInstance] shareToFriendZone];
    button.userInteractionEnabled = YES;
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: LinkAll/LinkAll/Other/Utils/ShareView.swift
//
//  ShareView.swift
//  LinkAll
//
//  Created by che on 15/11/23.
//  Copyright © 2015年 cheyongzi. All rights reserved.
//

import UIKit

class ShareView: UIView {
    var contentView: UIView!
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        self.addGestureRecognizer(UITapGestureRecognizer(target: self, action: "tapGestureAction"))
        
        self.backgroundColor = UIColor(white: 0.2, alpha: 0.5)
        
        let shareTitlesArray = [NSLocalizedString("Friends", comment: ""),NSLocalizedString("weChat", comment: ""),"Facebook","Twitter"]
        let shareImgArray = ["friends-icon","wechat-icon","facebook","twitter"]
        
        contentView = UIView(frame: CGRect(origin: CGPoint(x: 0, y: frame.size.height), size: CGSize(width: frame.size.width, height: 60)))
        contentView.backgroundColor = UIColor.whiteColor()
        addSubview(contentView)
        
        let length = (frame.size.width - 160) / 5
        
        for index in 0..<4 {
            let button = UIButton(type: .Custom)
            button.frame = CGRect(origin: CGPoint(x: length + CGFloat(index)*(40+length), y: 5), size: CGSize(width: 40, height: 40))
            button.backgroundColor = UIColor.clearColor()
            button.tag = index
            button.setImage(UIImage(named: shareImgArray[index]), forState: .Normal)
            button.addTarget(self, action: "buttonAction:", forControlEvents: .TouchUpInside)
            contentView.addSubview(button)
            
            let title = UILabel(frame: CGRect(origin: CGPointZero, size: CGSize(width: 50, height: 20)))
            title.center = CGPoint(x: button.center.x, y: 45)
            title.text = shareTitlesArray[index]
            title.backgroundColor = UIColor.clearColor()
            title.font = UIFont.systemFontOfSize(10)
            title.textAlignment = .Center
            contentView.addSubview(title)
        }
        
        UIView.animateWithDuration(0.2) {
            self.contentView.center = CGPoint(x: self.contentView.center.x, y: self.contentView.center.y-60)
        }
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func tapGestureAction() {
        UIView.animateWithDuration(0.2, animations: { () -> Void in
            self.contentView.center = CGPoint(x: self.contentView.center.x, y: self.contentView.center.y+60)
            }) { (finished: Bool) -> Void in
                if (finished) {
                    self.removeFromSuperview()
                }
        }
    }
    
    func buttonAction(sender: UIButton) {
        switch sender.tag {
        case 0:
            ShareUtils.shareInstance().shareToFriendZone()
        case 1:
            ShareUtils.shareInstance().shareToWeChat()
        case 2:
            ShareUtils.shareInstance().shareToFaceBook()
        case 3:
            ShareUtils.shareInstance().shareToTweeter()
        default:
            break
        }
    }

}
